util/chplenv/chpl_gmp.py

```diff
@@ -23,13 +23,18 @@ def get():
             gmp_subdir = os.path.join(third_party, 'gmp', 'install', uniq_cfg_path)
 
             if os.path.exists(os.path.join(gmp_subdir, 'include', 'gmp.h')):
-                gmp_val = 'gmp'
+                gmp_val = 'bundled'
             elif target_platform.startswith('cray-x'):
                 gmp_val = 'system'
             elif target_platform == 'aarch64':
                 gmp_val = 'system'
             else:
                 gmp_val = 'none'
+    elif gmp_val == 'gmp':
+        sys.stderr.write("Warning: CHPL_GMP=gmp is deprecated. "
```

Comment:
It would be nice to add a test for these messages in `test/deprecated`, maybe following the same pattern as `test/chplenv/printchplenv/printchplenv.chpl`, or at the very least, a `bundled.notest` that serves as a bread crumb to remind us to remove these deprecation warnings in later releases.